worker/worker.py

```diff
@@ -946,14 +946,28 @@ def pid_valid(pid, name):
                 subprocess.Popen(
                     ["ps", "-f", "-p", str(pid)],
```

Comment:
If I'm not wrong changing only these 2 lines should work for either linux or busybox:
```python
                subprocess.Popen(
                    ["ps", "-a", "-f"],

...

            if name in line and str(pid) in line:
                return True
```

Reply:
It seems you are right. I will change it. It is a bit strange though that `busybox --help` doesn't tell you this.
```
$ ps --help

BusyBox v1.35.0 (2022-05-09 17:27:12 UTC) multi-call binary.

Usage: ps [-o COL1,COL2=HEADER] [-T]

Show list of processes

	-o COL1,COL2=HEADER	Select columns for display
	-T			Show threads
```

Reply:
Neither here https://busybox.net/downloads/BusyBox.html

It seems that some common `ps` flags are accepted with null effect, perhaps for compatibility.

Reply:
By the way this could change in future...

Reply:
I don't think they are going to make ps less compatible.

Reply:
So I changed it.

Reply:
This is from the busybox ps source.

	// POSIX:
	// -a  Write information for all processes associated with terminals
	//     Implementations may omit session leaders from this list
	// -A  Write information for all processes
	// -d  Write information for all processes, except session leaders
	// -e  Write information for all processes (equivalent to -A)
	// -f  Generate a full listing
	// -l  Generate a long listing
	// -o col1,col2,col3=header
	//     Select which columns to display
	/* We allow (and ignore) most of the above. FIXME.
	 * -T is picked for threads (POSIX hasn't standardized it).
	 * procps v3.2.7 supports -T and shows tids as SPID column,
	 * it also supports -L where it shows tids as LWP column.
	 */

So at least -f -a are documented there. I wonder what the FIXME means.

Reply:
This comment was introduced in https://github.com/brgl/busybox/commit/9494919ea5116075e16daac1355265b023419cdc 16 years ago. I think the FIXME refers to the need for a more complete implementation (which was never done).

Reply:
It is strange since ``-p is also part of the posix spec. https://www.open-std.org/jtc1/sc22/open/n4217.pdf page 2976.

Reply:
Perhaps as you have pointed out it simply a very old copy & paste, at example busybox allows and ignores `ps -f p <pid_number>`, while exits with a warning for the more standard `-p` and `--pid`. Better to stick with the somehow documented `-f -a`

https://man7.org/linux/man-pages/man1/ps.1.html

```
       p pidlist
              Select by process ID.  Identical to -p and --pid.

       -p pidlist
              Select by PID.  This selects the processes whose process
              ID numbers appear in pidlist.  Identical to p and --pid.

       --pid pidlist
              Select by process ID.  Identical to -p and p.
```

Reply:
Correction: busybox accepts this command `ps -f abcdef 123456`, so the `p 1234` is simply not interpreted as a flag.

https://github.com/mirror/busybox/blob/master/procps/ps.c#L598
Just below the comment that you have highlighted, if I'm not wrong, there is the list of the flags accepted:
```
		getopt32(argv, "Zo:*aAdefl"IF_FEATURE_SHOW_THREADS("T"), &opt_o);
```